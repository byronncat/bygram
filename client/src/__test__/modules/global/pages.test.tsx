import { render, fireEvent } from '@testing-library/react'
import {
  BrowserRouter,
  createBrowserRouter,
  RouterProvider,
} from 'react-router-dom'
import rerender from 'react-test-renderer'
import '@testing-library/jest-dom'

import { ErrorPage } from '@global'

describe('pages', () => {
  it('should render the 404 page correctly', async () => {
    const errorComponent = (
      <BrowserRouter>
        <ErrorPage />
      </BrowserRouter>
    )

    const errorPage = rerender.create(errorComponent)
    expect(errorPage).toMatchSnapshot()

    const { getByText } = await render(errorComponent)
    expect(getByText(/404/i)).toBeInTheDocument()
    expect(getByText(/page not found/i)).toBeInTheDocument()
    expect(getByText(/back to home/i)).toBeInTheDocument()
  })

  test('can navigate to home page', async () => {
    window.history.pushState({}, '', '/bad/route')

    const router = createBrowserRouter([
      {
        path: '*',
        element: <ErrorPage />,
      },
      {
        errorElement: <ErrorPage />,
        children: [{ path: '/', element: <div>Homepage</div> }],
      },
    ])

    const { getByText } = await render(<RouterProvider router={router} />)

    const button = getByText(/back to home/i)
    expect(button).toBeInTheDocument()
    fireEvent.click(button)
    expect(window.location.pathname).toBe('/')
  })
})
