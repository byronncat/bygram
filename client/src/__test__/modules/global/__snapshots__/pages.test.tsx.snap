// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages should render the 404 page correctly 1`] = `
<div
  className="wrapper w-100 h-100 d-flex"
>
  <div
    className="square404"
    id="Square"
  >
    <span
      className="square flicker-1"
    >
      <p>
        404
      </p>
    </span>
  </div>
  <div
    className="texts d-flex flex-column justify-content-center"
  >
    <h4
      className="fs-4"
    >
      Oops! Page not found.
    </h4>
    <p
      className="pt-3 pb-5"
    >
      The page you are looking for does not exist. Go back to the main page.
    </p>
    <a
      className="button"
      href="/"
      onClick={[Function]}
    >
      Back to Home
    </a>
  </div>
</div>
`;
